### PR TITLE
[Bug #1326596] Show track title in window title bar

### PR DESCRIPTION
**Pull request by jercaianu**
https://bugs.launchpad.net/mixxx/+bug/1326596
The last added or played track is now showed as a title for the main window.

### CODE REVIEW

**Comment by badescunicu** [comment]
Please bear in mind that the number of decks is not fixed. Mixxx can handle more than two decks (there are skins who can display four decks), so we must find a solution which will work for any number of decks.

**Comment by jercaianu** [comment]
For this situation I could create a slot such as slotGetTrackFromDeck(int deck), however I don't know how to tie up a slot with a parameter in "connectValueChanged" from ControlObjectSlave.

**Comment by badescunicu** [comment]
Have you tried @ywwg's approach?
(```It'd be best to reuse the existing code that creates the history playlist -- when a track gets added to that list, the title bar can change.```)

This slot seems to be useful: https://github.com/mixxxdj/mixxx/blob/38dcd763c2ffc46e6958ca687a8a83e928059c4d/src/library/setlogfeature.h#L40

**Comment by ywwg** [comment]
> however I don't know how to tie up a slot with a parameter in "connectValueChanged" from ControlObjectSlave.

it is possible to do this with a QSignalMapper, but beware that they are really tricky to set up.  Take a look at what @badescunicu suggested, we already have code that tracks what the current-playing track is so you should use that.

**Comment by jercaianu** [comment]
Hello, I followed Nicu's suggestion and it seems to be working fine, however, the title is changing rather slow, due to the slow lookup in the sql database, I think.

File: src/library/setlogfeature.h [hunk 1]
```diff
@@ -47,6 +47,10 @@ class SetlogFeature : public BasePlaylistFeature {
     QAction* m_pJoinWithPreviousAction;
     QAction* m_pGetNewPlaylist;
     int m_playlistId;
+
+  signals:
+    void currentPlayingTrackChanged(QString r);
```
Comment (kain88-de) [inline src/library/setlogfeature.h]: Please give a meaningful name to the signal parameter or leave out the name

File: src/mixxx.h [hunk 2]
```diff
@@ -171,6 +174,7 @@ class MixxxMainWindow : public QMainWindow {
 
     void linkSkinWidget(ControlObjectSlave** pCOS,
                         ConfigKey key, const char* slot);
+    QString getTitleFromTrack(TrackPointer pTrack);
```
Comment (kain88-de) [inline src/mixxx.h]: please remove this function and the includes for it. You are never using it. I'm actually a little bit puzzled why this compiles at all.

Comment (ywwg) [inline src/mixxx.h]: c++!  if you declare a function in a header, but never use it, the compiler never has a chance to discover it doesn't exist.

**Comment by jercaianu** [comment]
Moved to pull request #549

**Pull request closed without merging**